vllm/v1/core/sched/scheduler.py

```diff
@@ -1085,6 +1085,29 @@ def update_from_output(
 
         return engine_core_outputs
 
+    def _update_computed_tokens(
+        self,
+        request: Request,
+        scheduled_spec_token_ids: list[int],
+        generated_token_ids: list[int],
+        spec_decoding_status: SpecDecodingStats | None,
+    ):
+        num_draft_tokens = len(scheduled_spec_token_ids)
+        num_accepted = len(generated_token_ids) - 1
+        num_rejected = num_draft_tokens - num_accepted
+        # num_computed_tokens represents the number of tokens
+        # processed in the current step, considering scheduled
+        # tokens and rejections. If some tokens are rejected,
+        # num_computed_tokens is decreased by the number of rejected
+        # tokens.
+        request.num_computed_tokens -= num_rejected
+        spec_decoding_stats = self.make_spec_decoding_stats(
+            spec_decoding_status,
+            num_draft_tokens=num_draft_tokens,
+            num_accepted_tokens=num_accepted,
+        )
+        return spec_decoding_stats
```

Comment:
I think duplication can be reduced here, perhaps keep this part outside of the method:
```python
        num_draft_tokens = len(scheduled_spec_token_ids)
        num_accepted = len(generated_token_ids) - 1
        num_rejected = num_draft_tokens - num_accepted
```

and then in the async_scheduler override, just update the placeholder count and then call
```python
return super()._update_computed_tokens(...)
```

Reply:
@Ronald1995 I was thinking of this being simplified further, so that the override in async_scheduling.py calls `return super()._update_computed_tokens(...)`.